lib/cloud_secret_env.rb

```diff
@@ -40,9 +40,9 @@ def provider
 
       def push_env!(secrets)
         block = if @config.override
-                  ->(k, v) { ENV[k] = v }
+                  ->(pair) { ENV[pair[0]] = pair[1] }
                 else
-                  ->(k, v) { ENV[k] = v unless ENV[k] }
+                  ->(pair) { ENV[pair[0]] = pair[1] unless ENV[pair[1]] }
```

Comment:
The unless should use `pair[0]` since it's the key value.

The goal is to set the environment variable if it isn't already set.

```suggestion
                  ->(pair) { ENV[pair[0]] = pair[1] unless ENV[pair[0]] }
```

Reply:
Good eye. I'll make that change and fix the tests to catch that.

Reply:
Fixed this and updated/added some tests to cover override true and false configs.